### DISCUSSION TITLE
Can't use Dbeaver on Rocky9

### DISCUSSION THREAD

**Discussion by jef66** [dbeaver/dbeaver#34699] [Q&A]
### Description

Hello,

I tried to use DBeaver on an fresh installation of Rocky9 with the RPM provided on the official website, but it didn't work. I have this kind of error : Could not load SWT library. (complete log in attachment).

[dbeaver_log.txt](https://github.com/dbeaver/dbeaver/files/15434215/dbeaver_log.txt)

I tried multiple version (the last one too, but not working).

Thanks for your help,

Regards

### DBeaver Version

23.3.5 and higher

### Operating System

Rocky9.3

### Steps to reproduce

1. install the RPM from official website
2. Just launch dbeaver-ce
3. Failed at start

**Comment by E1izabeth** [comment]
Thank you for bug report. Judging by the logs, he issue is the same is in #34034

**Comment by flambchap** [comment]
@jef66 It sounds like an issue with one of our dependencies. Can you please try the flatpak version and see if it runs?

**Comment by jef66** [comment]
Hello, same error with the flatpak version

**Comment by f-matt** [comment]
@jef66 , got it working on a fresh Rocky 9.4 install, with Dbeaver 24.1.0. Could you try upgrading your Rocky to 9.4?

**Comment by jef66** [comment]
I resintall the server with rocky 9.4 and still have the issue. Do you have installed any additionnal package or just the RPM ?
Is that possible to have the list of RPM package from your instance to check if I miss something ?

Thx

**Comment by f-matt** [comment]
I've installed Java 21 (java-21-openjdk), and then the Dbeaver RPM. You'll find atached all my installed packages (dnf list installed).
[installed-packages.txt](https://github.com/user-attachments/files/15812648/installed-packages.txt)

**Comment by E1izabeth** [comment]
Unfortunately, after upgrading to Eclipse 2024-06, seems that this OS are no longer supported, along with Windows 7, 8, macOS 10, and Windows Server 2012 and some UNIX OSes. This issue is not on DBeaver side, you can see the details here: https://github.com/eclipse-platform/eclipse.platform.swt/issues/1252.

**Comment by E1izabeth** [comment]
Here's the Eclipse supported target requirements https://eclipse.dev/eclipse/development/plans/eclipse_project_plan_4_32.xml#target_environments

**Comment by flambchap** [comment]
@E1izabeth The page you're referencing explicitly says that RHEL 9 is supported. Rocky Linux is bug-to-bug compatible with RHEL. It must be a different issue.

**Comment by E1izabeth** [comment]
However, the error is the same as for all unsupported OSes

**Comment by flambchap** [comment]
This error can happen for different reasons. For example, you'll see the same error if you install the latest Ubuntu on WSL2 and try to launch DBevaer there without installing any additional packages. That's because Ubuntu on WSL comes without libgtk. Installing something like Nautilus fixes the issue there. 

I have a gut feeling the issue can be resolved by installing some additional packages.

**Comment by flambchap** [comment]
But what's really odd is that the Flatpak version doesn't work either.

**Comment by flambchap** [comment]
I installed Rocky 9.4 aarch64 using Boot ISO, installed java 17 with `sudo dnf install java-17-openjdk`, installed DBeaver using the official tarball and everything worked fine.

**Comment by flambchap** [comment]
~~@jef66 Can you confirm that you use `java-17-openjdk` and not `java-17-openjdk-headless`?~~

Edit: Please disregard this; that's a stupid question. When it comes to RPMs, java is included in the package.

**Comment by jef66** [comment]
Hello, I tried on a fresh Rocky9.4 on my sandbox and it works.

Try again in my enterprise and failed. 
Tried with the tarball without Java included, then dnf install java-17-openjdk and it works !

Maybe be something on my server is blocking the loading of the library, I don't know, I will investigate.

Thx for your help !

**Comment by flambchap** [comment]
@jef66 I'm glad to hear that!

I suspect that some dependency of `java-17-openjdk` is a hard requirement for DBeaver, and it's not installed on your enterprise machine.

**Comment by michaeljmonte** [comment]
For people running in enterprise environments and have installed using the RPM, check in your /etc/fstab to see if your home directory mount has the `noexec` flag on it. If it does, you will get errors to the tune of "failed to map segment from shared object" for various libraries.

My solution was to download the .tar.gz version without java and run on a volume without the `noexec` flag.
